library/src/G4CMPKaplanQP.cc

```diff
@@ -433,6 +444,20 @@ G4CMPKaplanQP::CalcQPAbsorption(G4double qpE,
   return EDep;
 }
 
+
+// Handle quasiparticle energy-dependent absorption efficiency
+
+G4double G4CMPKaplanQP::CalcQPEfficiency(G4double qpE) const {
+    G4double eff = absorberEff + absorberEffSlope * qpE;
+
+  if (verboseLevel>2) {
+    G4cout << " CalcQPEfficiency " << eff << G4endl;
+  }
+
+  return std::min(eff, 1.);
```

Comment:
For the case of a negative slope, we ought to keep the efficiency non-negative as well:

`return std::max(0., std::min(eff, 1.));`